include/mata/number-predicate.hh

```diff
@@ -82,18 +88,53 @@ private:
     }
 
 public:
-    NumberPredicate(bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements) {};
+    //Trying to make NumberPredicate movable, does it make sense?
+    NumberPredicate(NumberPredicate&& rhs) = default;
+    NumberPredicate(const NumberPredicate& rhs) = default;
+    NumberPredicate & operator=(NumberPredicate&& rhs) = default;
+    NumberPredicate & operator=(const NumberPredicate& rhs) = default;
 
-    NumberPredicate(std::initializer_list <Number> list, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements) {
+    NumberPredicate(Number size,bool val,bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements), predicate(size, val)
+    {
+        if (tracking_elements && val) {
+            elements.reserve(size);
+            for (Number e = 0;e<size;++e)
+                elements.push_back(e);
+        }
+        if (!val)
+            cardinality = 0;
+        else
+            cardinality = size;
+    };
+
+    NumberPredicate(bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements), cardinality(0) {};
+
+    NumberPredicate(std::initializer_list <Number> list, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements), cardinality(0) {
         for (auto q: list)
             add(q);
     }
 
-    NumberPredicate(std::vector <Number> list, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements) {
+    NumberPredicate(std::vector <Number> list, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements), cardinality(0) {
         add(list);
     }
 
-    NumberPredicate(Mata::Util::OrdVector<Number> vec, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements) {
+    NumberPredicate(const std::vector<bool> & bv, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements), cardinality(0) {
+        predicate.reserve(bv.size());
+        for (size_t i = 0;i<bv.size();i++) {
+            if (bv[i])
+                add(i);
+        }
+    }
+
+    //just a hot fix
+    NumberPredicate(std::vector<bool> & bv, bool track_elements = true) : elements_are_exact(true), tracking_elements(track_elements), cardinality(0) {
```

Comment:
Hotfix of what? What purpose does it serve? Should it be deleted now, then? The comment should be improved if it is still relevant.